packages/eslint-plugin/docs/rules/no-unnecessary-condition.mdx

```diff
@@ -84,12 +84,68 @@ function bar(arg?: string | null) {
 
 {/* insert option description */}
 
-Example of correct code for `{ allowConstantLoopConditions: true }`:
+#### `'never'` or `false`
 
-```ts option='{ "allowConstantLoopConditions": true }' showPlaygroundButton
-while (true) {}
-for (; true; ) {}
-do {} while (true);
+Disallow constant conditions in loops.
+
+Example of incorrect code for `{ allowConstantLoopConditions: 'never' }`:
+
+```ts option='{ "allowConstantLoopConditions": "never" }' showPlaygroundButton
+while (true) {
+  // ...
+}
+
+for (; true; ) {
+  // ...
+}
+
+do {
+  // ...
+} while (true);
+```
+
+#### `'always'` or `true`
+
+Allow constant conditions in loops.
+
+Example of correct code for `{ allowConstantLoopConditions: 'always' }`:
+
+```ts option='{ "allowConstantLoopConditions": "always" }' showPlaygroundButton
+while (true) {
+  // ...
+}
+
+for (; true; ) {
+  // ...
+}
+
+do {
+  // ...
+} while (true);
+```
+
+#### `'neverExceptWhileTrue'`
```

Comment:
[Naming] Maybe... `onlyTrue` / `'only-true'` per https://github.com/typescript-eslint/typescript-eslint/issues/7047#issuecomment-1902126574?

I don't remember if there's every been a discussion on preferring camelCase vs. kebab-case for property values like this. 🤷

Reply:
Hmm yes, I agree that this is a lot cleaner 👍

I've done a quick search, and it seems all `enum` options are kebab-case, so I'm going with `only-true` to match this, thanks!